deploy/crds/crd-certificates.yaml

```diff
@@ -171,6 +171,64 @@ spec:
                 literalSubject:
                   description: "Requested X.509 certificate subject, represented using the LDAP \"String Representation of a Distinguished Name\" [1]. Important: the LDAP string format also specifies the order of the attributes in the subject, this is important when issuing certs for LDAP authentication. Example: `CN=foo,DC=corp,DC=example,DC=com` More info [1]: https://datatracker.ietf.org/doc/html/rfc4514 More info: https://github.com/cert-manager/cert-manager/issues/3203 More info: https://github.com/cert-manager/cert-manager/issues/4424 \n Cannot be set if the `subject` or `commonName` field is set. This is an Alpha Feature and is only enabled with the `--feature-gates=LiteralCertificateSubject=true` option set on both the controller and webhook components."
                   type: string
+                nameConstraints:
```

Comment:
what do you think about naming this field something like `caNameConstraints`? Or, if we think there may be *more* CA-only options that we may add in future (not sure?) under some kind of `caOptions` structure/field?

Reply:
I think `caNameConstraints` could be confusing (do these nameConstraints apply to the CA?).
Also, the validating webhook will guide users to no create certificates that do not make sense (eg. setting `isCA: false` and `nameConstraints`)

Reply:
nameConstraints as a setting is only valid for CA. Adding CA as prefix seems redundant. Also as @inteon suggested we have validation for nameConstraints to be used only if isCA is true. There is some similar code already used elsewhere where this is checked.

Reply:
The Certificate spec has 19 fields currently. I'd tend to agree that the API is getting big:

```console
$ k explain certificate.spec | grep '^   [a-z]' | cut -d'<' -f1 | sort
additionalOutputFormats
commonName
dnsNames
duration
emailAddresses
encodeUsagesInRequest
ipAddresses
isCA
issuerRef
keystores
literalSubject
privateKey
renewBefore
revisionHistoryLimit
secretName
secretTemplate
subject
uris
usages
```

But as a new user or advanced user, I would probably not look at the API reference anyways. I would probably look for the X.509 field they want in their Certificate resource and do a Google each with it, for example:

- "missing usages certificate cert-manager"
- "encode usages certificate cert-manager"
- "java keystore certificate cert-manager"

On top of that, we know that the X.509 spec is "bound" and isn't growing anymore, we can be confident that we will add a few fields in the future but not that many.

As long as we specify that `nameConstraints` is only usable when `isCA` is set to true, I am OK with using the name `nameConstraints` since it matches the name in the RFC.